disdrodb/data_transfer/download_data.py

```diff
@@ -0,0 +1,209 @@
+        Campaign name (eg :  EPFL_ROOF_2012).
+        If not provided (None), all campaigns will be downloaded.
+        The default is campaign_name=None.
+    station_names : str or list of str, optional
+        Station name.
+        If not provided (None), all stations will be downloaded.
+        The default is station_name=None.
+
+    Returns
+    -------
+    list
+        List of tuples containing the local path and the url.
+    """
+
+    # Get all config files from the metadata folders
+    list_of_base_path = []
+    if data_sources:
+        if isinstance(data_sources, str):
+            data_sources = [data_sources]
+    else:
+        data_sources = ["**"]
+    if campaign_names:
+        if isinstance(campaign_names, str):
+            campaign_names = [campaign_names]
+    else:
+        campaign_names = ["**"]
+
+    for data_source in data_sources:
+        for campaign_name in campaign_names:
+            base_path = os.path.join(disdrodb_dir, "Raw", data_source, campaign_name)
+            list_of_base_path.append(base_path)
+
+    metadata_folder_name = "metadata"
+
+    list_metadata_folders = []
+
+    for base_path in list_of_base_path:
+        if station_names:
+            if isinstance(station_names, str):
+                station_names = [station_names]
+            for station_name in station_names:
+                metadata_path = os.path.join(base_path, "**", metadata_folder_name, f"{station_name}.yml")
+                list_metadata_folders += glob.glob(metadata_path, recursive=True)
+        else:
+            metadata_path = os.path.join(base_path, "**", metadata_folder_name, "*.yml")
+            list_metadata_folders += glob.glob(metadata_path, recursive=True)
+
+    list_of_path_url = list()
+    for yaml_file_path in set(list_metadata_folders):
+        station_dir_path, data_url = get_station_local_remote_locations(yaml_file_path)
+        if data_url is None:
+            continue
+        list_of_path_url.append((station_dir_path, data_url))
+
+    return list_of_path_url
+
+
+def _download_file_from_url(url: str, dir_path: str, force: bool = False) -> None:
+    """Download file.
+
+    Parameters
+    ----------
+
+    url : str
+        URL of the file to download.
+    dir_path : str
+        Dir path where to download the file.
+    force : bool, optional
+        Overwrite the raw data file if already existing, by default False.
+
+    """
+
+    fname = os.path.basename(url)
+    file_path = os.path.join(dir_path, fname)
+
+    if not force and os.path.isfile(file_path):
+        print(f"{file_path} already exists, skipping download.")
+        return
+    elif force and os.path.isfile(file_path):
+        os.remove(file_path)
+
+    downloader = pooch.HTTPDownloader(progressbar=True)
+    pooch.retrieve(url=url, known_hash=None, path=dir_path, fname=fname, downloader=downloader, progressbar=tqdm)
+
+
+def _download_station_data(url_local_path: tuple, force: bool = False) -> None:

```

Comment:
Maybe would be worth having this function having as input `_download_station_data(metadata_fpath, force=False)` and put `station_dir_path, data_url = get_station_local_remote_locations(yaml_file_path)` inside it.

Reply:
ok